exchangeBridge/exchange.go

```diff
@@ -0,0 +1,161 @@
+package exchangeBridge
+
+import (
+	"context"
+
+	// We need the typeurl register calls that occur in the init of this package in order to be able to unmarshal events
+	// in the RepublishEvents func below
+	_ "github.com/containerd/containerd/api/events"
+
+	// Even though we are following the v2 runtime model, we are currently re-using a struct definition (Envelope) from
+	// the v1 event API
+	eventapi "github.com/containerd/containerd/api/services/events/v1"
+
+	"github.com/containerd/containerd/events"
+	"github.com/containerd/ttrpc"
+	"github.com/containerd/typeurl"
+	"github.com/gogo/protobuf/types"
+)
+
+const (
+	ExchangeBridgeServiceName = "aws.firecracker.containerd.exchangeBridge"
+	GetEventMethodName = "GetEvent"
+)
+
+// The ExchangeBridge interface provides an API for retrieving containerd events from a remote event exchange. It exists
+// separately from containerd's existing EventClient and EventServer interfaces in order to support retrieving events
+// via long-polling as opposed to a streaming model. This allows us to use TTRPC for the implementation, which currently
+// does not support streaming.
+type ExchangeBridge interface {
```

Comment:
I am curious if it makes some sense to expose `ExchangeBridgeService` and pass in a `EventListener` during creation. Need a better name I think for the `EventListener.

```go
type EventListener interface {
    GetEvent(ctx context.Context) (*eventapi.Envelope, error)
}

type TTRPCEventListener struct {
    // stuff
}

func NewExchangeBridgeService(ctx context.Context, eventSource events.Subscriber, e EventListener) {
}
```

Then you can have a `TTRPCEventListener`. This allows reuse of the of some of the code and allows switching out of how the events are retrieved. What are your thoughts on that?